### PR TITLE
Update to ghcup v0.1.14

### PR DESCRIPTION
**Pull request by Kleidukos**

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
             cabal: "2.4.0.1"
             expect-fail: false
           - os: ubuntu-latest
-            ghc: "8.12.0" # A version that will never exist.
+            ghc: "9.0.1" # A version that will never exist.
```
Comment (jared-w) [inline .github/workflows/workflow.yml:52]: 9.0.1 definitely exists :)

Perhaps 8.12.0 should be changed to something like 8.20.5 to be more clear that it's a fake version?

Comment (Kleidukos) [reply .github/workflows/workflow.yml:52]: uh, I don't understand how I managed not to parse that comment! :D

Comment (jared-w) [reply .github/workflows/workflow.yml]: Totally happens!

**Comment by jared-w** [comment]
Found the issue with windows:

- GHC 9 switched to hadrian by default for releases, I believe.
- Which still has a few kinks left in it. [Especially surrounding windows and packing components?](https://gitlab.haskell.org/ghc/ghc/-/issues/19571)
- [chocolatey currently works around this](https://github.com/Mistuke/GhcChoco/blob/master/9.0.1/ghc/tools/chocolateyInstall.ps1#L71) but missed `runhaskell` as an alias to `runghc`.

So it seems the temporary (permanent?) fix (for us) is to switch the CI to use `runghc` rather than `runhaskell` to test things out. I'm not sure creating an alias for runhaskell is the right thing for us to do. It seems like it would be very confusing for end users to have something installed by an action that installs something else that relies on upstream packaging yet it only works in the action? Even typing that out was quite confusing

tldr

```diff
--- a/.github/workflows/workflow.yml
+++ b/.github/workflows/workflow.yml
-      - name: Test runhaskell
+      - name: Test runghc
         run: |
-          runhaskell --version
-          runhaskell __tests__/hello.hs
+          runghc --version
+          runghc __tests__/hello.hs
```

should fix the CI, I believe.

**Comment by Kleidukos** [comment]
@jared-w it does indeed! Now I'm quite puzzled before the fact that a runner using the "fake version" (8.21.5) has been started by the matrix. Could you enlighten me on its purpose?

**Comment by jared-w** [comment]
> @jared-w it does indeed! Now I'm quite puzzled before the fact that a runner using the "fake version" (8.21.5) has been started by the matrix. Could you enlighten me on its purpose?

This is probably what I should've actually written the comment on :)

I have a fake version to test that fall through works correctly. I ran into problems before where a version wouldn't work on one method and it wouldn't try the others; this doesn't ensure all of them are tried, but it at least allows me to inspect that they all were.

I'd like to ideally have a test that actually made sure the fallbacks happened, but this got me most of the way there, so I left that for another day.

**Review by jared-w — APPROVED** [review]

**Comment by Kleidukos** [comment]
Ah, I see :)
Thanks for the explanation!
